### PR TITLE
Added LocalVariableRendering time and more information to timer Start+Tracing

### PR DESCRIPTION
**Pull request by DavidKarlas**
This allows for real time and precise tracing of debugger times

### CODE REVIEW

File: Mono.Debugging/Mono.Debugging.Client/DebuggerStatistics.cs [hunk 1]
```diff
@@ -39,6 +41,11 @@ public class DebuggerStatistics
 		TimeSpan minTime = TimeSpan.MaxValue;
 		TimeSpan maxTime, totalTime;
 
+		public DebuggerStatistics(string name)
+		{
+			Name = name;
+		}
```
Comment (mrward) [inline Mono.Debugging/Mono.Debugging.Client/DebuggerStatistics.cs:47]: API break? Should we also add the DebuggerStatistics method which takes no parameter?

File: Mono.Debugging/Mono.Debugging.Client/DebuggerStatistics.cs [hunk 2]
```diff
@@ -59,12 +66,11 @@ public class DebuggerStatistics
 		public int TimingsCount { get; private set; }
 
 		public int FailureCount { get; private set; }
+		public string Name { get; }
 
-		public DebuggerTimer StartTimer ()
+		public DebuggerTimer StartTimer (string name)
 		{
```
Comment (mrward) [inline Mono.Debugging/Mono.Debugging.Client/DebuggerStatistics.cs:72]: API break?

File: Mono.Debugging/Mono.Debugging.Client/DebuggerStatistics.cs [hunk 3]
```diff
@@ -136,19 +150,14 @@ public DebuggerTimer (DebuggerStatistics stats)
 			get { return stopwatch.Elapsed; }
 		}
 
-		public void Start ()
```
Comment (mrward) [inline Mono.Debugging/Mono.Debugging.Client/DebuggerStatistics.cs:139]: Another API break.

File: Mono.Debugging/Mono.Debugging.Client/DebuggerStatistics.cs [hunk 4]
```diff
@@ -120,10 +126,18 @@ public class DebuggerTimer : IDisposable
 		readonly DebuggerStatistics stats;
 		readonly Stopwatch stopwatch;
 
-		public DebuggerTimer (DebuggerStatistics stats)
+		static readonly TraceListener traceListener = new TraceSource (nameof (DebuggerTimer)).Listeners.OfType<TraceListener> ().FirstOrDefault (l => l.Name == nameof (DebuggerTimer));
```
Comment (mrward) [inline Mono.Debugging/Mono.Debugging.Client/DebuggerStatistics.cs:129]: Can we comment how to use this? I am assuming you need to setup a trace listener in VS Mac?

File: Mono.Debugging/Mono.Debugging.Client/DebuggerStatistics.cs [hunk 5]
```diff
@@ -120,10 +126,18 @@ public class DebuggerTimer : IDisposable
...
+		int traceId;
+
+		public DebuggerTimer (DebuggerStatistics stats, string name)
 		{
-			stopwatch = new Stopwatch ();
+			stopwatch = Stopwatch.StartNew ();
 			this.stats = stats;
+			if (stats != null && traceListener != null) {
+				traceId = Interlocked.Increment (ref traceSeq);
+				traceListener?.TraceEvent (null, stats.Name, TraceEventType.Start, traceId, name);
```
Comment (Therzok) [inline Mono.Debugging/Mono.Debugging.Client/DebuggerStatistics.cs:139]: Already null checked.

```suggestion
				traceListener.TraceEvent (null, stats.Name, TraceEventType.Start, traceId, name);
```

**Review by Therzok — APPROVED** [review]

**Review by mrward — COMMENTED** [review]
Code looks good to me. Do we have a problem with the public API changing? If not I can approve :)

**Review by mrward — APPROVED** [review]
Well, only VS Mac seems to use DebuggerStatistics. I suspect nobody else is using it.

**Comment by DavidKarlas** [comment]
About API break, yes, only VSMac is using it and for anyone in future I want to force them to use names for things... Hence removing old overload...
About how to use this, I will post in VSMac PR...

**Pull request merged**